src/ScalarMessaging.js

```diff
@@ -330,17 +332,54 @@ function setWidget(event, roomId) {
         name: widgetName,
         data: widgetData,
     };
-    if (widgetUrl === null) { // widget is being deleted
-        content = {};
-    }
 
-    client.sendStateEvent(roomId, "im.vector.modular.widgets", content, widgetId).done(() => {
+    if (userWidget) {
+        const client = MatrixClientPeg.get();
+        let userWidgets = {};
+        if (client.getAccountData('m.widgets')) {
```

Comment:
Why not use `Widgets.getUserWidgets()`?

Reply:
It needed a little bit of re-jigging to return in the right format. Done now.